Fix a bug where virtual attribute are included when aliasing joined table column names

finalwharf opened the pull request:
@FundingCircle/gdpr-engineering 👀

h-lame commented:
Seems fine based on everything you mentioned at standup - removing the names of the virtual attributes from the list of columns for the model sounds exactly like the sort of solution we need.  That said, I doubt I'd have the context to understand the fix if I was looking at this PR tomorrow or next week 'cos by then I'll have forgotten the details of our standup discussion.  To help future investigators can we include some more details of what the problem is and how it manifests itself in the commit message or PR description.  I think you mentioned that it's a bug with rails 4.x attribute API that was fixed in rails 5.x, so it would be great to include a link to that fix too if we can.

finalwharf commented:
I've updated the PR description with a proper explanation of the issue.
I've also changed the implementation a bit (see [L206](https://github.com/FundingCircle/fc-vault-rails/pull/28/files#diff-ba19de511d4dbf96b22d51bf1ecbf448R206))

h-lame commented:
Would it make sense to only include these fixes in AR 4.2?  E.g put a  `if ActiveRecord.version < Gem::Version("5")` guard around the fix in the first commit, and around the whole file in the 2nd commit?
